evm/vm/base.py

```diff
@@ -178,30 +178,32 @@ def import_block(self, block):
         ]
         if execution_data:
             headers, receipts, _ = zip(*execution_data)
+            header_with_txns = headers[-1]
         else:
-            headers, receipts = tuple(), tuple()
+            receipts = tuple()
+            header_with_txns = self.block.header
 
-        tx_root_hash, tx_kv_nodes = make_trie_root_and_nodes(block.transactions)
-        receipt_root_hash, receipt_kv_nodes = make_trie_root_and_nodes(receipts)
 
+        self.block = self.seal_block(self.block, header_with_txns, block.transactions, receipts)
+
+        return self.mine_block()
+
+    def seal_block(self, base_block, new_header, transactions, receipts):
```

Comment:
I don't like the term `seal` as for me at least, it implies mining like setting the `mix_hash` and `nonce`.

What about just `add_transaction_to_block` or something?

Reply:
How about `set_transactions_in_block`?
- pro: transactions is plural, old transactions in block are replaced, and -- for now -- the trie data is persisted
- con: implies block mutation, when it's really building a block copy

Other alternatives:
- `place_transactions_in_block`
- `commit_transactions_to_block`
- `get_block_with_transactions` (problem: get implies no side-effects)
- `add_transactions_to_block` (problem: add implies that previous transactions are left in place)

Reply:
- `set_block_transactions`  (keeps with `get_`, `set_` convention...)
- `update_block_transactions`  just an alternate prefix.

any of of those work for you?